vm/src/stdlib/os.rs

```diff
@@ -712,80 +671,58 @@ fn to_seconds_from_nanos(secs: i64, nanos: i64) -> f64 {
 }
 
 #[cfg(unix)]
-macro_rules! os_unix_stat_inner {
-    ( $path:expr, $follow_symlinks:expr, $vm:expr ) => {{
-        #[allow(clippy::match_bool)]
-        fn get_stats(path: &str, follow_symlinks: bool) -> io::Result<StatResult> {
-            let meta = match follow_symlinks {
-                true => fs::metadata(path)?,
-                false => fs::symlink_metadata(path)?,
-            };
-
-            Ok(StatResult {
-                st_mode: meta.st_mode(),
-                st_ino: meta.st_ino(),
-                st_dev: meta.st_dev(),
-                st_nlink: meta.st_nlink(),
-                st_uid: meta.st_uid(),
-                st_gid: meta.st_gid(),
-                st_size: meta.st_size(),
-                st_atime: to_seconds_from_unix_epoch(meta.accessed()?),
-                st_mtime: to_seconds_from_unix_epoch(meta.modified()?),
-                st_ctime: to_seconds_from_nanos(meta.st_ctime(), meta.st_ctime_nsec()),
-            })
-        }
-
-        get_stats($path.as_str(), $follow_symlinks.follow_symlinks)
-            .map_err(|err| convert_io_error($vm, err))
-    }};
-}
-
-#[cfg(target_os = "linux")]
 fn os_stat(
-    path: PyStringRef,
+    file: Either<PyStringRef, i64>,
```

Comment:
I think we should create a `PathLike` object as this will probably be used in many places.

Reply:
I really like that idea, but I think right now this construct is used in few enough places that it isn't worth it to extract it like that, and also for `os_stat` the code for `fd`s and paths is so different that it wouldn't really deduplicate that much.